test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -793,4 +785,142 @@ public void sealDoesNotModifyClientRouterEpoch() throws Exception {
+                        getManagementServer(SERVERS.PORT_2).getSequencerBootstrappedFuture()),
+                PARAMETERS.TIMEOUT_NORMAL
+        ).join();
+
+        addClientRule(corfuRuntime, SERVERS.ENDPOINT_0, new TestRule().always().drop());
+        layout.setEpoch(2L);
+        layout.moveServersToEpoch();
+        corfuRuntime.getLayoutView().updateLayout(layout, 1L);
+
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
+            Thread.sleep(PARAMETERS.TIMEOUT_SHORT.toMillis());
+            if (getLayoutServer(SERVERS.PORT_0).getCurrentLayout().equals(layout))
+                break;
+
+        }
+
+        assertThat(getLayoutServer(SERVERS.PORT_0).getCurrentLayout().getEpoch()).isEqualTo(2L);
+        assertThat(getLayoutServer(SERVERS.PORT_0).getCurrentLayout()).isEqualTo(layout);
+    }
+
+    /**
+     * A single node cluster is sealed but the client crashes before it can propose a layout.
+     * The management server now have to detect this state and fill up this slot with an existing
+     * layout in order to unblock the data plane operations.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void unblockSealedCluster() throws Exception {
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+        Layout l = (Layout) corfuRuntime.getLayoutView().getLayout().clone();
+
+        corfuRuntime.getRouter(SERVERS.ENDPOINT_0).getClient(ManagementClient.class)
+                .initiateFailureHandler().get();
+
+        CFUtils.within(
+                CompletableFuture.allOf(
+                        getManagementServer(SERVERS.PORT_0).getSequencerBootstrappedFuture()),
+                PARAMETERS.TIMEOUT_NORMAL
+        ).join();
+
+        l.setRuntime(corfuRuntime);
+        l.setEpoch(l.getEpoch() + 1);
+        l.moveServersToEpoch();
+
+        for (int i = 0; i < PARAMETERS.NUM_ITERATIONS_MODERATE; i++) {
+            Thread.sleep(PARAMETERS.TIMEOUT_SHORT.toMillis());
+            if (corfuRuntime.getLayoutView().getLayout().getEpoch() == l.getEpoch())
+                break;
+            corfuRuntime.invalidateLayout();
+        }
+        assertThat(corfuRuntime.getLayoutView().getLayout().getEpoch()).isEqualTo(l.getEpoch());
+    }
+
+    @Test
+    public void testNodeHealing() throws Exception {
+
+        addServer(SERVERS.PORT_0);
+        addServer(SERVERS.PORT_1);
+        addServer(SERVERS.PORT_2);
+        Layout l = new TestLayoutBuilder()
+                .setEpoch(1L)
+                .addLayoutServer(SERVERS.PORT_0)
+                .addLayoutServer(SERVERS.PORT_1)
+                .addLayoutServer(SERVERS.PORT_2)
+                .addSequencer(SERVERS.PORT_0)
+                .buildSegment()
+                .buildStripe()
+                .addLogUnit(SERVERS.PORT_0)
+                .addToSegment()
+                .addToLayout()
+                .addUnresponsiveServer(SERVERS.PORT_2)
```

Comment:
comment at this point as this is the main thing you are testing.

Reply:
Done